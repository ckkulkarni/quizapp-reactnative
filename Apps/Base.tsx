import React, {useState, useContext} from 'react';
import {
  Text,
  View,
  Button,
  TextInput,
  StyleSheet,
  ScrollView,
} from 'react-native';
import {useNavigation} from '@react-navigation/native';
import {NativeStackNavigationProp} from '@react-navigation/native-stack';
import SelectDropdown from 'react-native-select-dropdown';
import {QuizContext} from './QuizContext';
export default function Base() {
  const navigation = useNavigation<NativeStackNavigationProp<any>>();
  const {setName, setMail, setPhone, setSelected, setTracker} =
    useContext(QuizContext);
  const [validMail, setValid] = useState<boolean>(true);
  const [validName, setValidName] = useState<boolean>(false);
  const [validPhone, setValidPhone] = useState<boolean>(false);
  const [submitPressed, setSubmit] = useState<boolean>(false);
  const [validSelections, setSelections] = useState<boolean>(false);
  const [validOption, setOption] = useState<boolean>(false);
  const selections = ['ReactJS', 'React Native', 'JavaScript', 'Java'];
  function handleSubmit() {
    setSubmit(true);
  }
  const handleEmailChange = (text: string) => {
    setMail(text);
    const emailFormat = /^[A-Z0-9._%+-]+@[A-Z0-9.-]+\.[A-Z]{2,}$/i;
    setValid(emailFormat.test(text));
  };
  const handleNameChange = (text: string) => {
    setName(text);
    setValidName(text.trim().length > 0);
  };
  const handlePhoneChange = (text: string) => {
    setPhone(text);
    setValidPhone(text.trim().length === 10);
  };

  const isFormValid = validName && validMail && validPhone;
  const startQuiz = isFormValid && submitPressed;
  const track = [true, false];
  return (
    <ScrollView contentContainerStyle={styles.container}>
      <TextInput
        placeholder="Enter your name"
        onChangeText={handleNameChange}
        style={styles.input}
      />
      {!validMail && (
        <Text style={{color: 'red'}}>Enter a valid Email Address</Text>
      )}
      <TextInput
        placeholder="Enter your Email ID"
        onChangeText={handleEmailChange}
        keyboardType="email-address"
        style={[styles.input, !validMail && styles.invalidInput]}
      />
      <TextInput
        placeholder="Enter your Phone Number"
        keyboardType="numeric"
        onChangeText={handlePhoneChange}
        style={styles.input}
      />
      <Text>Select your preferred language: </Text>
      <SelectDropdown
        data={selections}
        onSelect={(selectedItem, index) => {
          setSelected(selectedItem);
        }}
        searchInputTxtColor={'blue'}
      />
      <Text>Track Score?</Text>
      <SelectDropdown
        data={track}
        onSelect={(tracking, index) => {
          setTracker(tracking);
        }}
        searchInputTxtColor={'blue'}
      />
      <Button title="Submit" onPress={handleSubmit} disabled={!isFormValid} />
      <Button
        disabled={!startQuiz}
        title="Click Here to Start Quiz"
        onPress={() => navigation.navigate('Question 1')}
      />
    </ScrollView>
  );
}
const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
  },
  input: {
    borderWidth: 1,
    borderColor: 'gray',
    borderRadius: 5,
    padding: 10,
    width: '80%',
    marginVertical: 10,
  },
  invalidInput: {
    borderColor: 'red',
  },
  selector: {
    backgroundColor: 'blue',
    padding: 10,
  },
});
