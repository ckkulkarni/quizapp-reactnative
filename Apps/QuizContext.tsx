import React from 'react';

export const QuizContext = React.createContext<{
  score: number;
  setScore: React.Dispatch<React.SetStateAction<number>>;
  name: string;
  setName: React.Dispatch<React.SetStateAction<string>>;
  email: string;
  setMail: React.Dispatch<React.SetStateAction<string>>;
  phone: string;
  setPhone: React.Dispatch<React.SetStateAction<string>>;
  selectedValue: string;
  setSelected: React.Dispatch<React.SetStateAction<string>>;
  trackScore: boolean;
  setTracker: React.Dispatch<React.SetStateAction<boolean>>;
}>({
  score: 0,
  setScore: () => {},
  name: '',
  setName: () => {},
  email: '',
  setMail: () => {},
  phone: '',
  setPhone: () => {},
  selectedValue: '',
  setSelected: () => {},
  trackScore: false,
  setTracker: () => {},
});
