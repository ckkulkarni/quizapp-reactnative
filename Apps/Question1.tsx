import React, {useContext, useState} from 'react';
import {
  Text,
  View,
  Button,
  Pressable,
  StyleSheet,
  TouchableOpacity,
} from 'react-native';
import {useNavigation} from '@react-navigation/native';
import {NativeStackNavigationProp} from '@react-navigation/native-stack/lib/typescript/src/types';
import {QuizContext} from './QuizContext';
export default function Question1() {
  const {trackScore, score, setScore, selectedValue} = useContext(QuizContext);
  const navigation = useNavigation<NativeStackNavigationProp<any>>();
  const [selectedIndex, setIndex] = useState<number>();
  const currentScore = score;
  const maxScore = 9;
  const [submitted, setSubmit] = useState<boolean>(false);
  let questionText = '';
  let options: any[] = [];
  if (selectedValue == 'ReactJS') {
    questionText = 'What is the extension used for TypeScript in ReactJS?';
    options = ['.tsx', '.jsx', '.js', '.java'];
  } else if (selectedValue == 'React Native') {
    questionText =
      'What are the simpler of the two CLIs used to create React Native Apps?';
    options = ['Expo CLI', 'React Native CLI'];
  } else if (selectedValue == 'JavaScript') {
    questionText = 'Inside which HTML element do we put the JavaScript?';
    options = ['<javascript>', '<script>', '<js>', '<scripting>'];
  } else if (selectedValue == 'Java') {
    questionText =
      'In Java, the class name should always be different from file name. True or False?';
    options = ['True', 'False'];
  }
  const handleOptionPress = (index: number) => {
    setIndex(index);
  };
  const handleAnswerCheck = () => {
    setSubmit(true);
    if (selectedValue == 'ReactJS') {
      if (selectedIndex === 0 && score < maxScore) {
        setScore(score + 1);
      } else {
        setScore(currentScore);
      }
    } else if (selectedValue == 'React Native' && score < maxScore) {
      if (selectedIndex === 0) {
        setScore(score + 1);
      } else {
        setScore(currentScore);
      }
    } else if (selectedValue == 'JavaScript' && score < maxScore) {
      if (selectedIndex === 1) {
        setScore(score + 1);
      } else {
        setScore(currentScore);
      }
    } else if (selectedValue == 'Java' && score < maxScore) {
      if (selectedIndex === 1) {
        setScore(score + 1);
      } else {
        setScore(currentScore);
      }
    }
  };

  return (
    <View style={styles.container} testID="questionContainer">
      <View style={styles.questionNavigation}>
        {[
          'Question 1',
          'Question 2',
          'Question 3',
          'Question 4',
          'Question 5',
        ].map((question, index) => (
          <TouchableOpacity
            key={index}
            onPress={() => navigation.navigate(question)}
            style={styles.navigationTouchable}
            testID={`navigate-to-${index}`}>
            <Text style={styles.navigationText}>{index + 1}</Text>
          </TouchableOpacity>
        ))}
      </View>
      {trackScore && <Text testID="score">Score: {score}</Text>}
      <Text style={{fontSize: 20, color: 'black'}} testID="questionID">
        {questionText}
      </Text>
      {options.map((option, index) => (
        <Pressable
          key={index}
          style={[
            styles.optionStyling,
            selectedIndex === index && styles.selectedOption,
          ]}
          onPress={() => handleOptionPress(index)}>
          <Text testID={`answer-${index}`}>{option.toString()}</Text>
        </Pressable>
      ))}
      <Button
        title="Answer"
        onPress={handleAnswerCheck}
        disabled={submitted}
        testID="answerButton"
      />
      {selectedIndex !== null && (
        <Button
          testID="question2"
          title="Next Question"
          onPress={() => navigation.navigate('Question 2')}
        />
      )}
    </View>
  );
}
const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
  },
  optionStyling: {
    backgroundColor: '#00A1E4',
    padding: 10,
    borderRadius: 20,
    marginTop: 6,
    marginBottom: 10,
    width: '70%',
  },
  selectedOption: {
    backgroundColor: '#CC3F0C',
  },
  questionNavigation: {
    display: 'flex',
    flexDirection: 'row',
    flexWrap: 'wrap',
    position: 'absolute',
    top: 0,
  },
  navigationTouchable: {
    backgroundColor: '#FEB95F',
    padding: 3,
    marginHorizontal: 7,
    borderWidth: 1,
    borderColor: 'black',
    borderRadius: 100,
    height: 30,
    width: 30,
    marginVertical: 20,
  },
  navigationText: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    marginLeft: 3,
  },
});
