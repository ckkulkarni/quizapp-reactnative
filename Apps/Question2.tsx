import React, {useContext, useState} from 'react';
import {
  Text,
  View,
  Button,
  Pressable,
  StyleSheet,
  TouchableOpacity,
} from 'react-native';
import {useNavigation} from '@react-navigation/native';
import {NativeStackNavigationProp} from '@react-navigation/native-stack/lib/typescript/src/types';
import {QuizContext} from './QuizContext';
export default function Question2() {
  const navigation = useNavigation<NativeStackNavigationProp<any>>();
  const {score, setScore, selectedValue, trackScore} = useContext(QuizContext);
  const [answerIndex, setIndex] = useState<number>();
  const currentScore = score;
  const maxScore = 9;
  const [submitted, setSubmit] = useState<boolean>(false);
  let question = '';
  let answers: any = [];
  if (selectedValue == 'ReactJS') {
    question = 'ReactJS a framework of JavaScript. True or false';
    answers = ['True', 'False'];
  } else if (selectedValue == 'React Native') {
    question =
      'React Native can only be used to create apps for Android and IOS. True or false?';
    answers = ['True', 'False'];
  } else if (selectedValue == 'JavaScript') {
    question = 'JavaScript must always be used for a web page. True or false?';
    answers = ['True', 'False'];
  } else if (selectedValue == 'Java') {
    question = 'JSP stands for Jakarta Server Pages. True or false?';
    answers = ['True', 'False'];
  }
  const handleAnswerCheck = () => {
    setSubmit(true);
    if (selectedValue == 'ReactJS') {
      if (answerIndex === 1 && score < maxScore) {
        setScore(score + 1);
      } else {
        setScore(currentScore);
      }
    } else if (selectedValue == 'React Native') {
      if (answerIndex === 0 && score < maxScore) {
        setScore(score + 1);
      } else {
        setScore(currentScore);
      }
    } else if (selectedValue == 'JavaScript') {
      if (answerIndex === 1 && score < maxScore) {
        setScore(score + 1);
      } else {
        setScore(currentScore);
      }
    } else if (selectedValue == 'Java') {
      if (answerIndex === 0 && score < maxScore) {
        setScore(score + 1);
      } else {
        setScore(currentScore);
      }
    }
  };

  const handleOptionPress = (index: number) => {
    setIndex(index);
  };
  return (
    <View style={styles.container} testID="questionContainer">
      <View style={styles.questionNavigation}>
        {[
          'Question 1',
          'Question 2',
          'Question 3',
          'Question 4',
          'Question 5',
        ].map((question, index) => (
          <TouchableOpacity
            testID={`goto-${index}`}
            key={index}
            onPress={() => navigation.navigate(question)}
            style={styles.navigationTouchable}>
            <Text style={styles.navigationText}>{index + 1}</Text>
          </TouchableOpacity>
        ))}
      </View>
      {trackScore && <Text testID="score">Score: {score}</Text>}
      <Text style={{fontSize: 20, color: 'black'}} testID="question">
        {question}
      </Text>
      {answers.map((answer: string, index: number) => (
        <Pressable
          key={index}
          style={[
            styles.optionStyling,
            answerIndex === index && styles.selectedOption,
          ]}
          onPress={() => handleOptionPress(index)}>
          <Text testID={`answer-${index}`}>{answer}</Text>
        </Pressable>
      ))}
      <Button
        title="Answer"
        onPress={handleAnswerCheck}
        disabled={submitted}
        testID="answerButton"
      />
      <Button
        title="Next Question"
        onPress={() => navigation.navigate('Question 3')}
        testID="nextQuestion"
      />
      <Button
        title="Go Back"
        onPress={() => navigation.goBack()}
        testID="previousQuestion"
      />
    </View>
  );
}
const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
  },
  optionStyling: {
    backgroundColor: '#00A1E4',
    padding: 10,
    borderRadius: 20,
    marginTop: 6,
    marginBottom: 10,
    width: '70%',
  },
  selectedOption: {
    backgroundColor: '#CC3F0C',
  },
  questionNavigation: {
    display: 'flex',
    flexDirection: 'row',
    flexWrap: 'wrap',
    position: 'absolute',
    top: 0,
  },
  navigationTouchable: {
    backgroundColor: '#FEB95F',
    padding: 3,
    marginHorizontal: 7,
    borderWidth: 1,
    borderColor: 'black',
    borderRadius: 100,
    height: 30,
    width: 30,
    marginVertical: 20,
  },
  navigationText: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    marginLeft: 3,
  },
});
