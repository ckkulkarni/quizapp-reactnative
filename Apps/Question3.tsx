import React, {useContext, useState} from 'react';
import {
  Text,
  View,
  Button,
  Pressable,
  StyleSheet,
  TouchableOpacity,
} from 'react-native';
import {useNavigation} from '@react-navigation/native';
import {NativeStackNavigationProp} from '@react-navigation/native-stack/lib/typescript/src/types';
import {QuizContext} from './QuizContext';
export default function Question3() {
  const navigation = useNavigation<NativeStackNavigationProp<any>>();
  const {score, trackScore, setScore, selectedValue} = useContext(QuizContext);
  const [answerIndex, setIndex] = useState<number>();
  const currentScore = score;
  const [submitted, setSubmit] = useState<boolean>(false);
  let question = '';
  const maxScore = 9;
  let answers: any = [];
  if (selectedValue == 'ReactJS') {
    question =
      'What does myReactApp refer to in the following command? npm create-react-app myReactApp';
    answers = [
      'Name of the App to be created',
      'The directory to create the app in',
      'The type of App to be created',
      'A reference to existing App',
    ];
  } else if (selectedValue == 'React Native') {
    question =
      'How can you create a new React Native project using React Native CLI?';
    answers = [
      'expo init',
      'react-native Project-name',
      'npx react-native init Project_Name',
      'expo react-native Project-name',
    ];
  } else if (selectedValue == 'JavaScript') {
    question =
      'What will the output of this code snippet: console.log(018 - 015)?';
    answers = ['NaN', '3', '5', '13'];
  } else if (selectedValue == 'Java') {
    question = 'What is a correct syntax to output "Hello World" in Java';
    answers = [
      "Console.WriteLine('Hello World')",
      "echo('Hello World')",
      "System.out.println('Hello World');",
      "print('Hello World')",
    ];
  }
  const handleAnswerCheck = () => {
    setSubmit(true);
    if (selectedValue == 'ReactJS') {
      if (answerIndex === 0 && score < maxScore) {
        setScore(score + 1);
      } else {
        setScore(currentScore);
      }
    } else if (selectedValue == 'React Native') {
      if (answerIndex === 2 && score < maxScore) {
        setScore(score + 1);
      } else {
        setScore(currentScore);
      }
    } else if (selectedValue == 'JavaScript') {
      if (answerIndex === 1 && score < maxScore) {
        setScore(score + 1);
      } else {
        setScore(currentScore);
      }
    } else if (selectedValue == 'Java') {
      if (answerIndex === 2 && score < maxScore) {
        setScore(score + 1);
      } else {
        setScore(currentScore);
      }
    }
  };

  const handleOptionPress = (index: number) => {
    setIndex(index);
  };
  return (
    <View style={styles.container} testID="questionContainer">
      <View style={styles.questionNavigation}>
        {[
          'Question 1',
          'Question 2',
          'Question 3',
          'Question 4',
          'Question 5',
        ].map((question, index) => (
          <TouchableOpacity
            key={index}
            onPress={() => navigation.navigate(question)}
            style={styles.navigationTouchable}
            testID={`goto-${index}`}>
            <Text style={styles.navigationText}>{index + 1}</Text>
          </TouchableOpacity>
        ))}
      </View>
      {trackScore && <Text>Score: {score}</Text>}
      <Text style={{fontSize: 20, color: 'black'}} testID="question">
        {question}
      </Text>
      {answers.map((answer: string, index: number) => (
        <Pressable
          key={index}
          style={[
            styles.optionStyling,
            answerIndex === index && styles.selectedOption,
          ]}
          onPress={() => handleOptionPress(index)}>
          <Text testID={`answer-${index}`}>{answer}</Text>
        </Pressable>
      ))}
      <Button
        title="Answer"
        onPress={handleAnswerCheck}
        disabled={submitted}
        testID="answerButton"
      />
      <Button
        title="Next Question"
        onPress={() => navigation.navigate('Question 4')}
        testID="nextQuestion"
      />
      <Button
        title="Go Back"
        onPress={() => navigation.goBack()}
        testID="previousQuestion"
      />
    </View>
  );
}
const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
  },
  optionStyling: {
    backgroundColor: '#00A1E4',
    padding: 10,
    borderRadius: 20,
    marginTop: 6,
    marginBottom: 10,
    width: '70%',
  },
  selectedOption: {
    backgroundColor: '#CC3F0C',
  },
  questionNavigation: {
    display: 'flex',
    flexDirection: 'row',
    flexWrap: 'wrap',
    position: 'absolute',
    top: 0,
  },
  navigationTouchable: {
    backgroundColor: '#FEB95F',
    padding: 3,
    marginHorizontal: 7,
    borderWidth: 1,
    borderColor: 'black',
    borderRadius: 100,
    height: 30,
    width: 30,
    marginVertical: 20,
  },
  navigationText: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    marginLeft: 3,
  },
});
