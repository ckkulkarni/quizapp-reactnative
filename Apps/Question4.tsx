import React, {useState, useContext} from 'react';
import {
  Text,
  View,
  Button,
  StyleSheet,
  Pressable,
  TouchableOpacity,
} from 'react-native';
import {useNavigation} from '@react-navigation/native';
import {NativeStackNavigationProp} from '@react-navigation/native-stack/lib/typescript/src/types';
import {QuizContext} from './QuizContext';
export default function Question4() {
  const [submitted, setSubmit] = useState<boolean>(false);
  const navigation = useNavigation<NativeStackNavigationProp<any>>();
  const {score, trackScore, setScore, selectedValue} = useContext(QuizContext);
  const [answerIndex, setIndex] = useState<number>();
  const maxScore = 9;
  const currentScore = score;
  let question = '';
  let answers: any = [];
  if (selectedValue == 'ReactJS') {
    question = 'What is the default port number in which the application run?';
    answers = ['3000', '8080', '3001', '5000'];
  } else if (selectedValue == 'React Native') {
    question = 'What is props?';
    answers = [
      'Which used for rendering dynamic changes whithin component',
      'Import component from one another component like react-native in app.js, custom component',
      'Which stands for properties and is being used for passing data from one component to another',
      'None of the above',
    ];
  } else if (selectedValue == 'JavaScript') {
    question =
      'How would you prompt users with messages and also accept inputs at the same time?';
    answers = ['Alert()', 'Display()', 'Prompt()', 'Confirm()'];
  } else if (selectedValue == 'Java') {
    question = 'What is the size of byte variable?';
    answers = ['8 bit', '16 bit', '32 bit', '64 bit'];
  }
  const handleAnswerCheck = () => {
    setSubmit(true);
    if (selectedValue == 'ReactJS') {
      if (answerIndex === 0 && score < maxScore) {
        setScore(score + 1);
      } else {
        setScore(currentScore);
      }
    } else if (selectedValue == 'React Native') {
      if (answerIndex === 2 && score < maxScore) {
        setScore(score + 1);
      } else {
        setScore(currentScore);
      }
    } else if (selectedValue == 'JavaScript') {
      if (answerIndex === 2 && score < maxScore) {
        setScore(score + 1);
      } else {
        setScore(currentScore);
      }
    } else if (selectedValue == 'Java') {
      if (answerIndex === 0 && score < maxScore) {
        setScore(score + 1);
      } else {
        setScore(currentScore);
      }
    }
  };
  const handleOptionPress = (index: number) => {
    setIndex(index);
  };
  return (
    <View style={styles.container} testID="questionContainer">
      <View style={styles.questionNavigation}>
        {[
          'Question 1',
          'Question 2',
          'Question 3',
          'Question 4',
          'Question 5',
        ].map((question, index) => (
          <TouchableOpacity
            key={index}
            onPress={() => navigation.navigate(question)}
            style={styles.navigationTouchable}>
            <Text style={styles.navigationText} testID={`goto-${index}`}>
              {index + 1}
            </Text>
          </TouchableOpacity>
        ))}
      </View>
      {trackScore && <Text>Score: {score}</Text>}
      <Text style={{fontSize: 20, color: 'black'}} testID="question">
        {question}
      </Text>
      {answers.map((answer: string, index: number) => (
        <Pressable
          key={index}
          style={[
            styles.optionStyling,
            answerIndex === index && styles.selectedOption,
          ]}
          onPress={() => handleOptionPress(index)}>
          <Text testID={`answer-${index}`}>{answer}</Text>
        </Pressable>
      ))}
      <Button
        title="Answer"
        onPress={handleAnswerCheck}
        disabled={submitted}
        testID="answerButton"
      />
      <Button
        title="Next Question"
        onPress={() => navigation.navigate('Question 5')}
        testID="nextQuestion"
      />
      <Button
        title="Go Back"
        onPress={() => navigation.goBack()}
        testID="previousQuestion"
      />
    </View>
  );
}
const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
  },
  optionStyling: {
    backgroundColor: '#00A1E4',
    padding: 10,
    borderRadius: 20,
    marginTop: 6,
    marginBottom: 10,
    width: '70%',
  },
  selectedOption: {
    backgroundColor: '#CC3F0C',
  },
  questionNavigation: {
    display: 'flex',
    flexDirection: 'row',
    flexWrap: 'wrap',
    position: 'absolute',
    top: 0,
  },
  navigationTouchable: {
    backgroundColor: '#FEB95F',
    padding: 3,
    marginHorizontal: 7,
    borderWidth: 1,
    borderColor: 'black',
    borderRadius: 100,
    height: 30,
    width: 30,
    marginVertical: 20,
  },
  navigationText: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    marginLeft: 3,
  },
});
