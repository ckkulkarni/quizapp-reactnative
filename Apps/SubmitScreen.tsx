import {Button, StyleSheet, Text, View} from 'react-native';
import React, {useContext} from 'react';
import {QuizContext} from './QuizContext';
import PieChart from 'react-native-pie-chart';
import {useNavigation} from '@react-navigation/native';
import {NativeStackNavigationProp} from '@react-navigation/native-stack/lib/typescript/src/types';
const SubmitScreen = () => {
  const navigation = useNavigation<NativeStackNavigationProp<any>>();
  const {score, setScore, selectedValue, name, email, phone} =
    useContext(QuizContext);
  const wrongAnswers = 9 - score;
  const chartData = [wrongAnswers, score];
  const chartColors = ['#A20021', '#B0FF92'];
  function restartQuiz() {
    setScore(0);
    navigation.navigate('Home');
  }
  return (
    <View style={styles.container} testID="scoreContainer">
      <View style={styles.pieStyling}>
        <PieChart
          widthAndHeight={200}
          series={chartData}
          sliceColor={chartColors}
        />
      </View>
      <View style={styles.statsView}>
        <Text style={styles.textStyle} testID="score">
          Score: {score}
        </Text>
        <Text style={styles.textStyle} testID="selectedStream">
          Quiz Stream: {selectedValue}
        </Text>
        <Text style={styles.textStyle} testID="name">
          Name: {name}
        </Text>
        <Text style={styles.textStyle} testID="mail">
          Mail: {email}
        </Text>
        <Text style={styles.textStyle} testID="phoneNo">
          Phone Number: {phone}
        </Text>
      </View>
      <View style={styles.buttonStyling}>
        <View style={{marginHorizontal: 10}}>
          <Button
            title="Re-enter Details"
            onPress={restartQuiz}
            testID="restartQuiz"
          />
        </View>
      </View>
    </View>
  );
};
export default SubmitScreen;
const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: 'white',
  },
  textStyle: {
    color: 'black',
    fontSize: 19,
    textAlign: 'center',
    backgroundColor: '#0496FF',
    padding: 5,
    marginTop: 15,
  },
  pieStyling: {
    position: 'absolute',
    top: 20,
  },
  statsView: {
    position: 'relative',
    top: 50,
  },
  buttonStyling: {
    display: 'flex',
    flexDirection: 'row',
    position: 'absolute',
    bottom: 30,
  },
});
