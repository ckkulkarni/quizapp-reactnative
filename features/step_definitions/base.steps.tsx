import {fireEvent, render, waitFor} from '@testing-library/react-native';
import 'react-native-get-random-values';
import {v4 as uuidv4} from 'uuid';
import Adapter from 'enzyme-adapter-react-16';
import {mount, shallow} from 'enzyme';
import * as enzyme from 'enzyme';
import Base from '../../Apps/Base';
import {loadFeature, defineFeature} from 'jest-cucumber';
import {QuizContext} from './../../Apps/QuizContext';
import {View, TextInput, Button} from 'react-native';
import {useNavigation} from '@react-navigation/native';
import {NativeStackNavigationProp} from '@react-navigation/native-stack';
const uuid = uuidv4();
const feature = loadFeature('./features/base.feature');
const navigation = useNavigation<NativeStackNavigationProp<any>>();

jest.mock('@react-navigation/native', () => {
  return {
    useNavigation: jest.fn().mockReturnValue({
      navigate: jest.fn(),
    }),
  };
});
enzyme.configure({adapter: new Adapter()});
defineFeature(feature, test => {
  const mockQuizContext = {
    score: 7,
    setScore: jest.fn(),
    name: 'chandu',
    setName: jest.fn(),
    email: 'chandra@chandu.com',
    setMail: jest.fn(),
    phone: '1234567890',
    setPhone: jest.fn(),
    selectedValue: 'ReactJS',
    setSelected: jest.fn(),
    trackScore: true,
    setTracker: jest.fn(),
  };

  test('User submits form with valid inputs', ({given, when, and, then}) => {
    let wrapper: any;
    beforeEach(() => {
      wrapper = shallow(<Base />);
    });
    given('I am on the base screen', () => {
      const wrapper = shallow(<Base />);
      expect(wrapper).toMatchSnapshot();
    });
    when('I enter a valid email "test@test.com"', () => {
      expect(wrapper.find('TextInput').at(0)).toBeTruthy();
    });
    and('I enter a valid name "John"', () => {
      expect(wrapper.find('TextInput').at(1)).toBeTruthy();
    });
    and('I enter a valid phone number "1234567890"', () => {
      expect(wrapper.find('TextInput').at(2)).toBeTruthy();
    });

    and('I submit the form', () => {
      const setSubmit = jest.fn();
      wrapper.find('TextInput').at(0).simulate('changeText', 'chandrakanth');
      wrapper
        .find('TextInput')
        .at(1)
        .simulate('changeText', 'chandra@kanth.com');
      wrapper.find('TextInput').at(2).simulate('changeText', '1234567890');
      wrapper.find('Button').at(0).props().onPress();
      expect(wrapper.find('Button').at(0).prop('disabled')).toBe(false);
    });

    then('I should see the start quiz button', () => {
      wrapper.find('Button').at(0).props().onPress();
      const startQuizButton = wrapper.find('Button').at(1);
      expect(startQuizButton.prop('disabled')).toBe(false);
    });
  });
});
