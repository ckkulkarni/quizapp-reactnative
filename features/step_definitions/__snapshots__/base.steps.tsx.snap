// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Quiz Start Button Navigation User submits form with valid inputs 1`] = `ShallowWrapper {}`;
